crates/sui-network/src/randomness/mod.rs

```diff
@@ -394,10 +398,19 @@ impl RandomnessEventLoop {
                 }) {
                 Ok(partial_sigs) => partial_sigs,
                 Err(e) => {
-                    debug!("failed to deserialize partial sigs: {e:?}");
+                    warn!("failed to deserialize partial sigs: {e:?}");
                     return;
                 }
             };
+        // Verify we received the expected share IDs (to protect against a validator that sends
+        // valid signatures of other peers which will be successfully verified below).
+        let received_share_ids = partial_sigs.iter().map(|s| s.index).collect::<Vec<_>>();
```

Comment:
does this need a sort? (and the "expected" Vec as well?) not sure anything in the protocol rn requires shares IDs to be in a particular order?

Reply:
I made sure that the crypto code returns those ordered, do you think we should order here to be on the safe side?

Reply:
it's fine then, maybe document in the `SendSignaturesRequest::partial_sigs` field that protocol requires sigs to be in fastcrypto provided order?